pkg/kube/ingress/ingressmgr.go

```diff
@@ -53,6 +53,15 @@ type Manager interface {
 // NewManager can be used to get the Manager based on the APIVersion of the ingress resources on the cluster
 // so that, respecitve methods from that APIVersion can be called
 func NewManager(ctx context.Context, kubeCli kubernetes.Interface) (Manager, error) {
+	// return early if its fake CLI because fakecli doesn't implement `ServerPreferredResources` correctly
+	list, err := kubeCli.Discovery().ServerPreferredResources()
+	if err != nil {
+		return nil, errors.New("Error finding if the CLI was fake CLI")
```

Comment:
```suggestion
		return nil, errors.Wrap(err, "Failed to discover the server preferred resources")
```
Not sure if we should mention fake cli in production code.